bokehjs/src/lib/models/formatters/datetime_tick_formatter.ts

```diff
@@ -164,40 +166,43 @@ export class DatetimeTickFormatter extends TickFormatter {
     const s0 = _strftime(t, this[resolution])
     const tm = _mktime(t)
     const resolution_index = resolution_order.indexOf(resolution)
-
+    let final_resolution = resolution
     let s = s0
-    let hybrid_handled = false
-    let next_index = resolution_index
-    let next_resolution = resolution
-
-    // As we format each tick, check to see if we are at a boundary of the
-    // next higher unit of time. If so, replace the current format with one
-    // from that resolution. This is not the best heuristic but it works.
-    while (tm[tm_index_for_resolution.get(resolution_order[next_index])!] == 0) {
-      next_index += 1
-
-      if (next_index == resolution_order.length)
-        break
-
-      // The way to check that we are at the boundary of the next unit of
-      // time is by checking that we have 0 units of the resolution, i.e.
-      // we are at zero minutes, so display hours, or we are at zero seconds,
-      // so display minutes (and if that is zero as well, then display hours).
-      if ((resolution == "minsec" || resolution == "hourmin") && !hybrid_handled) {
-        if ((resolution == "minsec" && tm[4] == 0 && tm[5] != 0) || (resolution == "hourmin" && tm[3] == 0 && tm[4] != 0)) {
-          next_resolution = resolution_order[resolution_index-1]
-          s = _strftime(t, this[next_resolution])
+
+    if (this.scale_up_boundary) {
+      let hybrid_handled = false
+      let next_index = resolution_index
+      let next_resolution = resolution
+
+      // As we format each tick, check to see if we are at a boundary of the
+      // next higher unit of time. If so, replace the current format with one
+      // from that resolution. This is not the best heuristic but it works.
+      while (tm[tm_index_for_resolution.get(resolution_order[next_index])!] == 0) {
+        next_index += 1
+
+        if (next_index == resolution_order.length)
           break
-        } else {
-          hybrid_handled = true
+
+        // The way to check that we are at the boundary of the next unit of
+        // time is by checking that we have 0 units of the resolution, i.e.
+        // we are at zero minutes, so display hours, or we are at zero seconds,
+        // so display minutes (and if that is zero as well, then display hours).
+        if ((resolution == "minsec" || resolution == "hourmin") && !hybrid_handled) {
+          if ((resolution == "minsec" && tm[4] == 0 && tm[5] != 0) || (resolution == "hourmin" && tm[3] == 0 && tm[4] != 0)) {
+            next_resolution = resolution_order[resolution_index-1]
+            s = _strftime(t, this[next_resolution])
+            break
+          } else {
+            hybrid_handled = true
+          }
         }
-      }
 
-      next_resolution = resolution_order[next_index]
-      s = _strftime(t, this[next_resolution])
+        next_resolution = resolution_order[next_index]
+        s = _strftime(t, this[next_resolution])
+      }
+      final_resolution = next_resolution
```

Comment:
I'll definitely have to study this a bit, it's been many years since I've dug into the tick formatting. I am a little surprised that the `scale_up_boundary  = false` case is evidently just a no-op, it seems like it would need to still do all this logic, with the exception of skipping the `hybrid_handled` bits